snuba/optimize.py

```diff
@@ -145,72 +169,141 @@ def _subdivide_parts(
     return output
 
 
-def optimize_partition_runner(
+def concurrent_optimize_partition_runner(
     clickhouse: ClickhousePool,
     database: str,
     table: str,
     parts: Sequence[util.Part],
-    ignore_cutoff: bool,
     parallel: int,
+    cutoff_time: Optional[datetime],
+    optimize_partition_tracker: Optional[OptimizedPartitionTracker],
     clickhouse_host: Optional[str] = None,
 ) -> None:
-    if parallel <= 1:
-        optimize_partitions(
-            clickhouse, database, table, parts, ignore_cutoff, clickhouse_host
+    """
+    Run optimize job concurrently using multiple threads.
+
+    It subdivides the partitions into number of threads which are needed to be run. Each thread then runs the optimize
```

Comment:
nit: PEP8 suggest 72 charactes limit per line on docstring to make them easier to read in code reviews 
https://peps.python.org/pep-0008/#maximum-line-length